### PR TITLE
Fix name shadowing

### LINKED ISSUE #94: Build with -Wshadow
Upgrade Cable to 0.4.0 which enables `-Wshadow`. Then fix compilation issues.

### PR DESCRIPTION
**Pull request by chfast**
Fixes #94

### CODE REVIEW

File: include/intx/intx.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ struct uint
 
     constexpr uint() noexcept = default;
 
-    constexpr uint(half_type hi, half_type lo) noexcept : lo(lo), hi(hi) {}
+    constexpr uint(half_type high, half_type low) noexcept : lo(low), hi(high) {}
```
Comment (axic) [inline include/intx/intx.hpp]: Why not `_hi` and _lo`?

Comment (chfast) [reply include/intx/intx.hpp]: Looks horrible in documentation and any IDE assistant.

Comment (axic) [reply include/intx/intx.hpp]: I don't mind was just curious as that is a recommendation in many places/languages nowadays.

Comment (chfast) [reply include/intx/intx.hpp]: I try to avoid additional rules just to have less rules. I believe the `_` prefix was mostly required in cpp-ethereum.

Comment (axic) [reply include/intx/intx.hpp]: Solidity uses it too and now Rust as well.

Comment (chfast) [reply include/intx/intx.hpp]: cpp-ethereum -> solidity.

**Pull request merged**